lib/board.rb

```diff
@@ -16,4 +16,31 @@ def initialize
   def valid_coordinate?(coordinate)
     @cells.key?(coordinate)
   end
+
+  def valid_placement?(ship, coordinates)
```

Comment:
This looks great! We might be able to refactor down the road to condense it a bit and make it more readable. Maybe through the use of a helper method or two